scenario/uploaddata.go

```diff
@@ -108,86 +102,47 @@ func (settings UploadDataSettings) Execute(
 		actionState.AddErrors(errors.Wrapf(err, "failed to open file <%s>", settings.Filename))
 		return
 	}
-	defer func() {
-		err = file.Close()
-		if err != nil {
-			actionState.AddErrors(errors.Wrapf(err, "failed to close file <%s>", settings.Filename))
-		}
-	}()
 
-	body := helpers.GlobalBufferPool.Get()
-	defer helpers.GlobalBufferPool.Put(body)
-	writer := multipart.NewWriter(body)
-
-	// Then create the binary multipart field
-	part, err := writer.CreateFormFile("data", filepath.Base(settings.Filename))
+	tempFileClient, err := tempcontent.NewTUSClient(sessionState, connection, defaultChunkSize, 5)
 	if err != nil {
-		actionState.AddErrors(errors.Wrapf(err, "failed to create multipart form"))
+		actionState.AddErrors(errors.WithStack(err))
 		return
 	}
-	_, err = io.Copy(part, file)
+	tempFile, err := tempFileClient.UploadFromFile(sessionState.BaseContext(), file)
```

Comment:
Since I guess total upload time could take some time I understand why we don't use the new context function which would have the timeout set by `timeout` script parameter, but also not sure we should try uploading until failed or entire test times out. Maybe add an optional upload timeout flag (default set high like 1h our so), and create context  with this? This would be different from the retrybackoff.

Reply:
not sure, opinions?

Reply:
Added an opt-in timeout to uploaddata and uploadapp